src/modules/Channel/components/ChannelUI/index.tsx

```diff
@@ -44,7 +62,10 @@ const ChannelUI = (props: ChannelUIProps) => {
       isInvalid={isInvalid}
       renderChannelHeader={(props) => (<ChannelHeader {...props} />)}
       renderMessageList={(props) => (<MessageList {...props} />)}
-      renderMessageInput={() => (<MessageInputWrapper {...props} />)}
+      renderMessageInput={() => (
+        props.renderMessageInput?.()
+        ?? <MessageInputWrapper {...props} disabled={disableMessageInput} />
+      )}
```

Comment:
It looks like the code here could use some cleanup. It seems that `renderChannelHeader` might be overridden as well. It would be beneficial to handle these changes in a separate pull request.
@HoonBaek Would you like to give it a try? 🌝

Reply:
oh yes thanks, i forgot to mention this. renderMessageList and renderChannelHeader also need to handle props

Reply:
Hoon will handle this as a separate PR.